temporian/core/operators/tick.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2021 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Tick operator class and public API function definitions."""
+
+from temporian.core import operator_lib
+from temporian.core.data.node import Node, create_node_new_features_new_sampling
+from temporian.core.operators.base import Operator
+from temporian.proto import core_pb2 as pb
+from temporian.core.data.duration import (
+    Duration,
+    NormalizedDuration,
+    normalize_duration,
+)
+
+
+class Tick(Operator):
+    def __init__(
+        self, input: Node, interval: NormalizedDuration, rounding: bool
+    ):
+        super().__init__()
+
+        self._interval = interval
+        self._rounding = rounding
+
+        self.add_input("input", input)
+        self.add_attribute("interval", interval)
+        self.add_attribute("rounding", rounding)
+
+        self.add_output(
+            "output",
+            create_node_new_features_new_sampling(
+                features=[],
+                indexes=input.schema.indexes,
+                is_unix_timestamp=input.schema.is_unix_timestamp,
+                creator=self,
+            ),
+        )
+
+        self.check()
+
+    @property
+    def interval(self) -> NormalizedDuration:
+        return self._interval
+
+    @property
+    def rounding(self) -> bool:
+        return self._rounding
+
+    @classmethod
+    def build_op_definition(cls) -> pb.OperatorDef:
+        return pb.OperatorDef(
+            key="TICK",
+            attributes=[
+                pb.OperatorDef.Attribute(
+                    key="interval",
+                    type=pb.OperatorDef.Attribute.Type.FLOAT_64,
+                ),
+                pb.OperatorDef.Attribute(
+                    key="rounding",
+                    type=pb.OperatorDef.Attribute.Type.BOOL,
+                ),
+            ],
+            inputs=[pb.OperatorDef.Input(key="input")],
+            outputs=[pb.OperatorDef.Output(key="output")],
+        )
+
+
+operator_lib.register_operator(Tick)
+
+
+def tick(input: Node, interval: Duration, rounding: bool = True) -> Node:
+    """Generates timestamps at regular intervaled in the range of a guide.
+
+    Args:
+        input: Guide node. The start and end time boundaries to generate the new
+            timestamps are defined by the range of timestamps in `input`.
```

Comment:
I think it could be useful (and easy to implement) to allow receiving two different nodes for the start and end dates? `tick(start: Node, end: Node, interval: Duration, ...`. 

Would be nice to have the second one be optional (in which case we use the first for both) but then we would need to have it as the third parameter after `interval` which might look weird